### PR TITLE
EncounterManager

### PR DESCRIPTION
**Pull request by Spike-Zbb**

### CODE REVIEW

File: Assets/Scripts/EncounterScripts/EncounterManager.cs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+		{
+			Debug.LogWarning("There is no ENCOUNTER DATA");
+		}
+	}
+
+	/// <summary>
+	/// Finish Combat
+	/// </summary>
+	/// <param name="outcome"></param>
+	public void OnEncounterComplete(EncounterOutcome outcome)
```
Comment (caterunions) [inline Assets/Scripts/EncounterScripts/EncounterManager.cs:41]: is encounter outcome as an enum necessary? i feel like "bool victory" would accomplish the same thing as we're never gonna have a third outcome

File: Assets/Scripts/EncounterScripts/Encounter.cs [hunk 2]
```diff
@@ -0,0 +1,12 @@
+using UnityEngine;
+
+public class Encounter : MonoBehaviour 
+{
+	public string id;
+	public GameObject[] enemyGroup;
```
Comment (caterunions) [inline Assets/Scripts/EncounterScripts/Encounter.cs:6]: serialize private versions of these and expose read-only public ones

File: Assets/Scripts/EncounterScripts/TileEncounterTrigger.cs [hunk 3]
```diff
@@ -0,0 +1,32 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class TileEncounterTrigger : MonoBehaviour
+{
+	public Encounter encounter;
+	private EncounterManager encounterManager;
```
Comment (caterunions) [inline Assets/Scripts/EncounterScripts/TileEncounterTrigger.cs:8]: make encounter private and serialize it for the inspector, also follow variable naming conventions:
private -> _variable
public -> Variable

File: Assets/Scripts/EncounterScripts/TileEncounterTrigger.cs [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+	private void Start()
+	{
+		encounterManager = FindObjectOfType<EncounterManager>();
```
Comment (caterunions) [inline Assets/Scripts/EncounterScripts/TileEncounterTrigger.cs:12]: serialize a reference to encounter manager if possible, avoid find

File: Assets/Scripts/EncounterScripts/Encounter.cs [hunk 5]
```diff
@@ -0,0 +1,12 @@
+using UnityEngine;
+
+public class Encounter : MonoBehaviour 
+{
+	public string id;
```
Comment (caterunions) [inline Assets/Scripts/EncounterScripts/Encounter.cs:5]: id is never used, and also should ideally be stored as something other than a string (if we need it...? unsure of its purpose)

**Review by caterunions — CHANGES_REQUESTED** [review]
requesting changes on the access level of some variables, and naming conventions to match the code currently in the project.

**Comment by Icewav3** [comment]
@Spike-Zbb waiting on your response

**Comment by Spike-Zbb** [comment]
Hi Cole, sorry, I just got home. I have already finished editing it. Do you want me to upload it now?

________________________________
发件人: Cole Dorman ***@***.***>
发送时间: 2024年11月23日 18:29
收件人: Icewav3/TacticalGS3 ***@***.***>
抄送: Yuhang Zhang ***@***.***>; Mention ***@***.***>
主题: Re: [Icewav3/TacticalGS3] EncounterManager (PR #19)


CAUTION: This email originates from outside RRC. Please verify the sender and always use caution with any requests, links or attached documents.


@Spike-Zbb<https://github.com/Spike-Zbb> waiting on your response

―

**Comment by Spike-Zbb** [comment]
I uploaded the latest changes

**Review by caterunions — APPROVED** [review]

**Comment by caterunions** [comment]
All the changes look great, thanks!

File: Assets/Scripts/EncounterScripts/Encounter.cs [hunk 6]
```diff
@@ -0,0 +1,12 @@
+using UnityEngine;
+
+public class Encounter : MonoBehaviour 
+{
+	[SerializeField]
+	private GameObject[] enemyGroup;
+
+	public void StartEncounter()
+	{
+		Debug.Log("StartEncounter");
```
Comment (Icewav3) [inline Assets/Scripts/EncounterScripts/Encounter.cs:10]: Use the sceneGod for this
example:  SceneGod.SInstance.EnterCombatState(enemyGroup);

File: Assets/Scripts/EncounterScripts/EncounterManager.cs [hunk 7]
```diff
@@ -0,0 +1,68 @@
+using UnityEngine;
+
+public class EncounterManager : MonoBehaviour
```
Comment (Icewav3) [inline Assets/Scripts/EncounterScripts/EncounterManager.cs:3]: not sure if we can really utilize this class to its potential, as most of its functionality is in the sceneGod. Maybe just call thigns in the TileEncounterTrigger

File: Assets/Scripts/EncounterScripts/TileEncounterTrigger.cs [hunk 8]
```diff
@@ -0,0 +1,26 @@
+using UnityEngine;
+
+public class TileEncounterTrigger : MonoBehaviour
+{
+	[SerializeField]
+	private Encounter _encounter;
```
Comment (Icewav3) [inline Assets/Scripts/EncounterScripts/TileEncounterTrigger.cs:6]: this should probably be enemyGroup

File: Assets/Scripts/EncounterScripts/TileEncounterTrigger.cs [hunk 9]
```diff
@@ -0,0 +1,26 @@
+using UnityEngine;
+
+public class TileEncounterTrigger : MonoBehaviour
```
Comment (Icewav3) [inline Assets/Scripts/EncounterScripts/TileEncounterTrigger.cs:3]: I think encounter can be safely merged into this without causing problems

**Review by Icewav3 — CHANGES_REQUESTED** [review]

**Comment by Spike-Zbb** [comment]
I re-modified the code according to Cole's suggestion, but I am not sure whether the _**SceneGod.SInstance.EnterCombatState(enemyGroup)**_ method needs parameters.

**Comment by Spike-Zbb** [comment]
What does LevelEncounters do? When should I call the method in this ScriptableObject?

**Pull request merged**